staging/src/k8s.io/apiserver/pkg/cel/library/cost.go

```diff
@@ -278,7 +295,7 @@ func (l *CostEstimator) EstimateCallCost(function, overloadId string, target *ch
 	case "url":
 		if len(args) == 1 {
 			sz := l.sizeEstimate(args[0])
-			return &checker.CallEstimate{CostEstimate: sz.MultiplyByCostFactor(common.StringTraversalCostFactor)}
+			return &checker.CallEstimate{CostEstimate: sz.MultiplyByCostFactor(common.StringTraversalCostFactor), ResultSize: &sz}
```

Comment:
is it a bug to construct CallEstimate and omit ResultSize? we do that pretty much everywhere else in this function, right?

Reply:
It's not.  It's only needed for variable length types (lists, maps, strings or types that have a underlying unbounded length string representation).

Reply:
is  ResultSize needed here because a big URL input propagates to a big URL output, rather than the no-return or boolean-return values from other functions?

Reply:
That's right.  If we want to be able to estimate the "size" of a URL for estimated cost for equals, we need to propagate the result size.  All of our other custom types don't need that-- once they parse a string input, the internal representation can be used for a bunch of O(1) ops.